README.md

```diff
@@ -4,4 +4,128 @@ Logs HTTP files to an S3 bucket at a configurable frequency.
+      "producers": [
+        "polling_producer_name",
+        "webhook_producer_name"
+      ]
+    }
+  }
+}
+```
+
+## Producers
+
+Delta produces files from two sources: polled HTTP endpoints and webhooks.
+
+### Polling
+
+Polling sources make an HTTP request for data at a configurable interval.
+
+Configuration options:
+* `url`: Required.
+* `frequency`: Default 60000.
+* `headers`: Default `{}`. Example: `{"content-type": "application/x-protobuf"}`.
+* `filters`: Default `[]`. See Filters below.
+
+### Webhooks
+
+Webhooks accept an HTTP POST to `/webhook/:webhook_producer_name`, and treat the body of
+the request as the content of the file.
+
+Configuration options:
+* `type`: Must be `"webhook"`.
+* `authorization`: A string. Default `null`. If present, checks that incoming requests have an `"authorization"` header set to this value.
+* `filters`: Default `[]`. See Filters below.
+
+### Filters
+
+Both polling and webhook sources also accept filters, which can arbitrarily
+process the File into 0 or more Files. Some example filters:
+
+- ensure the body is GZip-compressed (or not compressed)
+- convert a JSON body into multiple Files based on an access path
+- rename the File based on an access path
+- set the `updated_at` value based on an access path
+
+Each entry in the configuration is a list whose first element is the string name of a function in [`lib/delta/file.ex`](lib/delta/file.ex), and any further elements are arguments to pass to that function.
+
+Example:
+```json
+"filters": [
+  ["ensure_not_encoded"],
+  ["json_updated_at", ["metadata", "timestamp"]],
+]
+```
+
+All producers will always finish with `["ensure_content_type"]` and `["ensure_gzipped"]`.
+
+## Sinks
+
+Sinks take the generated Files and write them somewhere.
+
+### S3
+
+The most useful sink writes the Files to an Amazon S3 bucket.
+
+Writes to the given bucket at the path `{prefix}/{year}/{month}/{day}/{time}_{url}.gz`.
+
+Configuration options:
+* `type`: must be `"s3"`.
+* `bucket`: Required string.
+* `prefix`: Default `""`. Will prepend this to all file names it writes to s3.
+* `acl`: Default `"public-read"`. Passed to `S3.put_object`.
+* `producers`: Required list of string producer names.
+
+### Log
+
+Useful for debugging, the Log sink logs a message for each File.
+
+Configuration options:
+* `type`: must be `"log"`.
+* `producers`: Required list of string producer names.
+
+## Headers
+
+Delta uses the `content-type` and `content-encoding` headers. If the `content-type` is missing, it can fall back to the known extensions listed in `config/config.exs`, with a default of `application/octet-stream`. The only supported `content-encoding` is `"gzip"`.
+
+## Installation
+
+```
+asdf install
+mix deps.get
+env DELTA_JSON="$(cat config.json)" mix phx.server
+```
+
+By default, Delta loads configuration the `DELTA_JSON` environment
+variable. You can see a sample JSON configuration in
+[priv/default_configuration.json](priv/default_configuration.json).
```

Comment:
At the moment, this isn't exactly right:
- with `MIX_ENV=dev` (default), `DELTA_JSON` will be checked first, and if it doesn't exist, `priv/default_configuration.json` will be used.
- with `MIX_ENV=prod` (or in a release), only `DELTA_JSON` will be used.